src/DataStructures/Python/TensorData.cpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "DataStructures/Tensor/TensorData.hpp"
+#include <memory>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "PythonBindings/BoundChecks.hpp"
+#include "Utilities/GetOutput.hpp"
+
+namespace py = pybind11;
+
+namespace py_bindings {
+void bind_tensordata(py::module& m) {  // NOLINT
+  // Wrapper for TensorComponent
+  py::class_<TensorComponent>(m, "TensorComponent")
+      .def(py::init<std::string, DataVector>(), py::arg("name"),
+           py::arg("data"))
+      .def("set_name",
+           [](TensorComponent& tc, const std::string& n) { tc.name = n; })
+      .def("get_name", [](const TensorComponent& tc) { return tc.name; })
```

Comment:
We try to mirror the C++ interface as closely as possible in Python bindings. Therefore please expose the `name` and `data` fields as attributes which is how they are exposed in the C++ code, not through accessor functions. Just use `def_readwrite`: https://pybind11.readthedocs.io/en/stable/classes.html#instance-and-static-fields. The same applies to `ExtentsAndTensorVolumeData` below.